include/circt/Dialect/FIRRTL/CHIRRTLOps.td

```diff
@@ -40,7 +40,8 @@ def CombMemOp : CHIRRTLOp<"combmem", [HasCustomSSAName, DeclareOpInterfaceMethod
   let arguments = (ins StrAttr:$name, NameKindAttr:$nameKind,
                        AnnotationArrayAttr:$annotations,
                        OptionalAttr<InnerSymAttr>:$inner_sym,
-                       OptionalAttr<MemoryInitAttr>:$init);
+                       OptionalAttr<MemoryInitAttr>:$init,
+                       OptionalAttr<StrAttr>:$prefix);
```

Comment:
This can represent `OptionalAttr<StrAttr>:$prefix` as `DefaultValuedAttr<StrAttr, "{}">`. I thought there used to be some tradeoff here. Is there or are these the same thing? (Is there any cost to one or the other?)

Reply:
Yeah these are a little different, I quickly did some testing for three different options here:
- `OptionalAttr` allows the attribute to be `null`, builders still require it as an argument, `getPrefix()` returns an `optional<StringRef>`
- `DefaultValuedAttr` does not allow the attr to be null, generates a builder which does not require it as an argument, `getPrefix()` returns a `StringRef`
- `DefaultValuedOptionalAttr` allows the attr to be null, builders still require it as an argument, `getPrefix()` returns `StringRef` by using the default value if the attr is null

I used `OptionalAttr` here because that is what is used for `MemOp` and `FirMemOp`.

I don't really know what is best.  Using an optional attribute is kind of confusing because we end up allowing both empty string prefix ("") and null string prefix, and its not clear what the difference is.  We have, in the past, decided not to use Optional attrs when there is a empty value for the attribute, to avoid this kind of confusion, but I know that this can have real overhead costs (but memories are relatively rare).  I see that we use OptionalAttr for `inner_sym`, despite it having a empty value.